dewy/common/collection_embeddings.py

```diff
@@ -0,0 +1,259 @@
+from typing import List, Self, Tuple
+
+import asyncpg
+from llama_index.embeddings import OpenAIEmbedding
+from llama_index.node_parser import SentenceSplitter
+from llama_index.schema import TextNode
+from loguru import logger
+
+from dewy.chunks.models import TextChunk
+from dewy.collections.models import DistanceMetric
+from dewy.collections.router import get_dimensions
+
+from .extract import extract
+
+
+class CollectionEmbeddings:
+    """Helper class for working with the embeddings in a collection."""
+
+    def __init__(self, pg_pool: asyncpg.Pool, collection_row: asyncpg.Record) -> None:
```

Comment:
Sort of weird that this object is responsible for interpreting the record - might be cleaner to pass through collection id, name etc, and pull them out of the record first. Would also make this easier to test, potentially.

Reply:
Yeah. I thought so too, but it made more sense before. Now, I think I can move it into the `classmethods`. Generally, I think it would make sense to cache the embeddings rather than always creating, but this was a "hey, can do a single query for document and the needed information about the collection" type thing. Moved into the `from_{document,collection}_id` methods.